service/devdoc/registry_requirements.md

```diff
@@ -370,17 +370,234 @@ Request-Id: <guid>
+```
+  GET /twins/<encodeURIComponent(deviceId)>/modules/<encodeURIComponent(moduleId)>?api-version=<version> HTTP/1.1
+  Authorization: <config.sharedAccessSignature>
+  Request-Id: <guid>
+```
+**]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_003: [** The `getModuleTwin` method shall call the `done` callback with a `Twin` object updated with the latest property values stored in the IoT Hub service. **]**
+
+### updateModuleTwin(deviceId: string, moduleId: string, patch: any, etag: string, done: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_004: [** The `updateModuleTwin` method shall throw a `ReferenceError` exception if `deviceId`, `moduleId`, `patch`, `etag`,or `done` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_005: [** The `updateModuleTwin` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+PATCH /twins/<encodeURIComponent(deviceId)/modules/<encodeURIComponent(moduleId)>>?api-version=<version> HTTP/1.1
+Authorization: <config.sharedAccessSignature>
+Content-Type: application/json; charset=utf-8
+Request-Id: <guid>
+If-Match: <etag>
+<patch>
+```
+**]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_006: [** The `updateModuleTwin` method shall call the `done` callback with a `Twin` object updated with the latest property values stored in the IoT Hub service. **]**
+
+## Device configuration
+
+### addConfiguration(configuration: Configuration, done: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_007: [** The `addConfiguration` method shall throw a `ReferenceError` exception if `configuration` or `done` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_008: [** The `addConfiguration` method shall throw an `ArgumentError` exception if `configuration.id` is falsy. **]**
+**SRS_NODE_IOTHUB_REGISTRY_18_009: [** The `addConfiguration` method shall set `configuration.schemaVersion` to the constant `currentConfigurationSchemaVersion` if it is not already set. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_010: [** The `addConfiguration` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+PUT /configurations/<encodeURIComponent(configuration.id)>?api-version=<version> HTTP/1.1
+Authorization: <sharedAccessSignature>
+Content-Type: application/json; charset=utf-8
+Request-Id: <guid>
+
+<configuration>
+```
+**]**
+
+### getConfiguration(configurationId: string, done: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_011: [** The `getConfiguration` method shall throw a `ReferenceError` exception if `configurationId` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_012: [** The `getConfiguration` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+GET /configurations/<encodeURIComponent(configurationId)>?api-version=<version> HTTP/1.1
+Authorization: <sharedAccessSignature>
+Request-Id: <guid>
+```
+**]**
+
+### getConfigurations(done: Registry.ResponseCallback): void;
```

Comment:
This seems like an odd api.  There is no limit on the number of configurations returned.  It would seem as though a query interface would be better...

Reply:
Yeah it seems at odds with what the service has been pushing for devices and modules (deprecating the "get multiple entries" API in favor of queries). is there a limit to the number of configurations that can be registered at a time? 

---
In reply to: [176860487](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176860487) [](ancestors = 176860487)